### PR TITLE
Implement fallback to deviceinfo if getprop fails

### LINKED ISSUE #9: Provide alternative device type determination for Non-Android devices
Currently system-image calls getprop to determine the device type. This is not supported on Non-Android phones, and needs a replacement.

### PR DESCRIPTION
**Pull request by Flohack74**
This PR was remade to have the right branch name. This is needed for Non-Android phones that do not have getprop implemented. Should fix #9

### CODE REVIEW

**Comment by nanu-c** [comment]
Works on my side, it's also verified with push server and it needs a reboot to pick up new device name

**Comment by mariogrip** [comment]
A better idea would be to replace it fully with deviceinfo and place it into the xenial_-_edge_-_wayland branch

**Comment by nanu-c** [comment]
But then we have again multiple branches instead of 1 master branch or is the xenial_-edge-_wayland the one and only master and this here is going to be dropped?

**Comment by Flohack74** [comment]
@nanu-c Marius offered this as a possibiliity to get it merged before OTA-12 is released since we are freezing there (as much as we can, since still things gets merged to some repos it seems)

**Comment by Flohack74** [comment]
Hmm what we are going to do with this? I can see Pinephones now reporting correctly with deviceinfo tag. Shall we close this @mariogrip

File: systemimage/device.py [hunk 1]
```diff
@@ -38,13 +38,17 @@ class SystemProperty(BaseDevice):
...
-                'getprop ro.product.device'.split(), universal_newlines=True)
-        except CalledProcessError as error:
-            log.exception('getprop exit status: {}'.format(error.returncode))
-            return '?'
-        except FileNotFoundError as error:
-            log.exception('getprop command not found')
-            return '?'
-        return stdout.strip()
+            stdout = check_output('getprop ro.product.device'.split(), universal_newlines=True).strip()
+        except:
```
Comment (dobey) [inline systemimage/device.py:44]: Please do not use blanket exceptions. Instead this should be `except CalledProcessError:` to only catch the failure to run getprop.

File: systemimage/device.py [hunk 2]
```diff
@@ -38,13 +38,17 @@ class SystemProperty(BaseDevice):
...
+            pass
+        if stdout == '':
+            # Try to use device-info instead
+            try:
+                stdout = check_output('device-info get name'.split(), universal_newlines=True).strip()
+                log.info(stdout)
+            except:
```
Comment (dobey) [inline systemimage/device.py:51]: Another blanket except clause. Please catch the appropriate `CalledProcessError` instead.

**Review by dobey — CHANGES_REQUESTED** [review]

**Comment by peat-psuwit** [comment]
Migrated to GitLab.

https://gitlab.com/ubports/core/system-image/-/merge_requests/11

**Pull request closed without merging**